### DISCUSSION TITLE
issue: useFieldArray values are not working properly with Controller

### DISCUSSION THREAD

**Discussion by konarssuresh** [react-hook-form/react-hook-form#10412] [Q&A]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-controller-issue-forked-2dm88j?file=/src/index.js

### Steps to reproduce

1. Go to 'index page'
2. Click on 'APPEND'
3. type something in first name and last name inputs
4. Click on PREPEND
5. Click on SUBMIT

You can see correct values in form values but controller does not update that

![image](https://github.com/react-hook-form/react-hook-form/assets/29543196/71bdab5d-afad-48d7-ad27-e19ecabe484b)

Please let me know if there is any mistake on my part.

### Expected behaviour

* New inputs should be added at the start of fieldArray
* Errors should only be shown for empty inputs.

This issue seems to have occurred between versions `7.26.0` and `7.26.1`.

you can see that same code works fine in 7.21.0 - 
https://codesandbox.io/s/usefieldarray-controller-issue-7-21-0-9kkehm?file=/src/index.js

Please let me know if there is any mistake made from my side

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sujinleeme** [comment]
Hello, I noticed that you use `key={index}` not `item.id`, which is a unique field identifier. It functions well now. 
Check this [codesandbox](https://codesandbox.io/p/sandbox/usefieldarray-controller-issue-forked-lwhgzz?file=%2Fsrc%2FFieldArrayComp.jsx%3A20%2C12).
